### ISSUE TITLE
be less chatty about failed to parse

### ISSUE THREAD

**Issue by rzezeski** [oxidecomputer/opte#148]
Now that more people are running Omicron + OPTE I think it makes sense to not be so chatty in the logs without some opt-in. For example, the way OPTE/xde currently work allows more traffic than you really want into the OPTE engine. Because OPTE still has a lot of IPv6 support to implement it just punts on things in various places. One example is not even recognizing ICMPv6:

```
WARNING: failed to parse packet: BadHeader("IPv6: UnexpectedNextHeader { next_header: 58 }")
WARNING: failed to parse packet: BadHeader("IPv6: UnexpectedNextHeader { next_header: 58 }")
```

I'd rather not have user's system logs filled with these messages, possibly distracting from more pressing issues (and just being annoying and not very useful in general). I think it makes more sense to put this behind `opte_debug` for those that want max squeaky wheel. Furthermore, there is `bad-packet` probe to track this stuff as well.

**Comment by rzezeski** [comment]
Added DTrace script to trace the bad packets probes:

```
root@sled1:~/dtrace# ./opte-trace opte-bad-packet
PORT         DIR MBLK               MSG
unknown      in  0xfffffe03963809a0 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396373d80 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396373d80 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe038f7c3740 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380c60 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 UnsupportedProtocol(ICMPv6)
PORT         DIR MBLK               MSG
unknown      in  0xfffffe0396380060 BadHeader("IPv6: UnexpectedNextHeader { next_header: 0 }")
unknown      in  0xfffffe038a35c4e0 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe038a35c4e0 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396380060 BadHeader("IPv6: UnexpectedNextHeader { next_header: 0 }")
unknown      in  0xfffffe0396380060 BadHeader("IPv6: UnexpectedNextHeader { next_header: 0 }")
unknown      in  0xfffffe038a35c4e0 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe038d734ee0 UnsupportedProtocol(ICMPv6)
unknown      in  0xfffffe0396373f00 Rx bad packet: no outer header
PORT         DIR MBLK               MSG
unknown      in  0xfffffe038d734460 BadHeader("IPv6: UnexpectedNextHeader { next_header: 0 }")
unknown      in  0xfffffe0396373f00 Rx bad packet: no outer header
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
unknown      in  0xfffffe03963807e0 BadHeader("IPv6: UnexpectedNextHeader { next_header: 0 }")
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
unknown      in  0xfffffe03963807e0 BadHeader("IPv6: UnexpectedNextHeader { next_header: 0 }")
PORT         DIR MBLK               MSG
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
unknown      in  0xfffffe0396380aa0 Rx bad packet: no outer header
```

**Comment by rzezeski** [comment]
Addressed in 978527fec717f3d732642f7eab210d539b6622cb.
